### PR TITLE
leaflet: Implement Evented and GeoJSON interface

### PR DESCRIPTION
**Pull request by lediur**
Went through the Leaflet 1.0 rc3 docs and transcribed the Evented and GeoJSON docs into this interface. Also added descriptions because why not?

### CODE REVIEW

File: leaflet/index.d.ts [hunk 1]
```diff
@@ -199,8 +199,146 @@ declare namespace L {
 
     export function bounds(points: BoundsLiteral): Bounds;
 
+    /**
+     * A set of methods shared between event-powered classes (like Map and Marker).
+     * Generally, events allow you to execute some function when something happens
+     * with an object (e.g. the user clicks on the map, causing the map to fire
+     * 'click' event).
+     */
     export interface Evented {
```
Comment (alejo90) [inline leaflet/index.d.ts]: All of the methods returning `Evented` should return `this`

**Comment by andy-ms** [comment]
Same as #11431, it would be better to target `master` and incorporate into #11445.

File: leaflet/index.d.ts [hunk 2]
```diff
@@ -476,6 +614,245 @@ declare namespace L {
     export function circle(latlng: LatLngLiteral, radius: number, options?: CircleOptions): Circle;
 
     export function circle(latlng: LatLngTuple, radius: number, options?: CircleOptions): Circle;
+    
+    /**
+     * Used to group several layers and handle them as one.
+     * If you add it to the map, any layers added or removed from the group will be
+     * added/removed on the map as well. Extends Layer.
+     */
+    export interface LayerGroup extends Layer {
```
Comment (alejo90) [inline leaflet/index.d.ts]: All of the methods returning `LayerGroup` should return `this`

File: leaflet/index.d.ts [hunk 3]
```diff
@@ -476,6 +614,245 @@ declare namespace L {
...
+    /**
+     * Create a feature group, optionally given an initial set of layers.
+     */
+    export function featureGroup(layers?: Layer[]): FeatureGroup;
+
+    type StyleFunction = (feature: any) => PathOptions;
+
+
+    export interface GeoJSONOptions {
+        /**
```
Comment (alejo90) [inline leaflet/index.d.ts]: I believe all of the fields on this option are optional.

**Review by alejo90 — COMMENTED** [review]
Looks great, there are issues with the return types, though. Using the actual return type instead of `this` will prevent the fluent interface to work correctly on child types.

File: leaflet/index.d.ts (hunk 3 above)
Comment (lediur) [inline leaflet/index.d.ts]: Yep, that's right. Thanks for catching this!

**Pull request closed without merging**